### DISCUSSION TITLE
Burgers Vector: Error using 'weights' with calcDensity

### DISCUSSION THREAD

**Discussion by DriptD** [mtex-toolbox/mtex#2058] [Ask Anything] [answered]
Hello,

I am looking to get a weighted distribution of the Burgers vector, for a single phase, using the code below (code copied from @ralfHielscher's answer [here](https://github.com/mtex-toolbox/mtex/discussions/1930#discussioncomment-6813959))

```
ebsdSGrid=ebsdS.gridify                        % Prepare the EBSD data 
W=weightedBurgersVec(ebsdSGrid)      % Compute the Weighted Burger Vectors (WBV)
WC= inv(ebsdSGrid.orientations).* W    % Transforms WBV into crystal coordinates
WDistri = calcDensity(WC,'weights',norm(W));      % Compute weighted density of WC

plot(WDistri)
mtexColorMap('hot)
mtexColorBar
```
However, the following error pops up if I pass `'weights',norm(W)` into `calcDensity()` (works fine without the `'weights'`)

```
Arrays have incompatible sizes for this operation.

Error in S2FunHarmonic.quadrature (line 90)
  nfsftmex('set_f', plan, W(:) .* values(:, index));

Error in Miller/calcDensity (line 36)
sF = 4*pi * S2FunHarmonic.quadrature(h,ones(size(h)),varargin{:});
```
I have checked `size(WC)` and `size(W)`. They are identical. Strangely, `calcDensity()` also works fine with `'weights'` if I plot pole figures weighted by grain sizes.

I would really appreciate if anyone could provide a solution to this. I am using Mtex 5.10.2 and Matlab 2021a on a Windows 11 PC.

Thanks!

Sincerely,
Dripta

**Comment by kilir** [comment, marked as answer]
Hi,
getting rid of the nan values should do the trick.
```
Wn = norm(W(:));
ind = isnan(Wn);
WDistri = calcDensity(WC(ind),'weights',Wn(ind));  
```

Cheers,
Rüdiger

**Reply by DriptD** [reply]
Hello Rüdiger,

It works with `ind = ~isnan(Wn)`. Thanks a million! 

Sincerely,
Dripta
